zkvm/src/txlog.rs

```diff
@@ -76,6 +84,73 @@ impl TxID {
         entry.commit_to_transcript(&mut t);
         t.challenge_bytes(b"merkle.leaf", result);
     }
+
+    /// Computes the Merkle proof of inclusion required to validate an
+    /// entry at index i, not including the original item i.
+    /// Fails when requested index is out of bounds
+    pub fn proof(list: &[Entry], index: usize) -> Result<Vec<MerkleHash>, VMError> {
+        if index >= list.len() {
+            return Err(VMError::InvalidMerkleProof);
+        }
+        let t = Transcript::new(b"ZkVM.txid");
+        let mut result = Vec::new();
+        Self::subproof(t, list, index, &mut result)?;
+        Ok(result)
+    }
+
+    fn subproof(
+        t: Transcript,
+        list: &[Entry],
+        index: usize,
+        result: &mut Vec<MerkleHash>,
+    ) -> Result<(), VMError> {
+        match list.len() {
+            0 => Err(VMError::InvalidMerkleProof),
```

Comment:
seems like this case is already checked in the `proof`, allowing us to make `subproof` non-failable.